server/src/main/java/org/opensearch/index/shard/CheckpointRefreshListener.java

```diff
@@ -38,8 +38,8 @@ public void beforeRefresh() throws IOException {
 
     @Override
     public void afterRefresh(boolean didRefresh) throws IOException {
-        if (shard.routingEntry().primary()) {
-            publisher.publish(shard.getLatestReplicationCheckpoint());
+        if (shard.routingEntry().primary() && shard.indexSettings().isSegrepEnabled()) {
```

Comment:
We conditionally wire up the refreshListener in IndexShard if segrep is enabled & the shard is a primary, I'm not sure we need these checks anymore?

Reply:
Sure, i can remove this. Follow up question - should we be checking `didRefresh` here before publishing the checkpoint ?

Reply:
good call, yes I think we should.  Otherwise we could make multiple calls for the same ReplicationCheckpoint.